### PR TITLE
Use commontestdata and refactor testLeoRoutes

### PR DESCRIPTION
**Pull request by vkumra-broad**
Refactored CommonTestData and TestLeoRoutes, moved the common code to CommonTestData.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala [hunk 1]
```diff
@@ -23,16 +22,16 @@ import scala.concurrent.duration._
...
   implicit val routeTimeout = RouteTestTimeout(10 seconds)
 
   val clusterName = "test"
   val googleProject = "dsp-leo-test"
   val unauthorizedTokenCookie = HttpCookiePair("LeoToken", "unauthorized")
   val expiredTokenCookie = HttpCookiePair("LeoToken", "expired")
-  val serviceAccountEmail = WorkbenchEmail("pet-1234567890@test-project.iam.gserviceaccount.com")
-  val userEmail = WorkbenchEmail("user1@example.com")
+  //  val serviceAccountEmail = WorkbenchEmail("pet-1234567890@test-project.iam.gserviceaccount.com")
+  //  val userEmail = WorkbenchEmail("user1@example.com")
```
Comment (kyuksel) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala]: Should the commented-out lines above be removed?

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/TestLeoRoutes.scala [hunk 2]
```diff
@@ -76,13 +44,13 @@ trait TestLeoRoutes { this: ScalatestRouteTest with ScalaFutures with Matchers =
...
-  def notebookServiceAccount(googleProject: GoogleProject): Option[WorkbenchEmail] = {
-    serviceAccountProvider.getNotebookServiceAccount(defaultUserInfo, googleProject).futureValue
-  }
+//  def clusterServiceAccount(googleProject: GoogleProject): Option[WorkbenchEmail] = {
+//    serviceAccountProvider.getClusterServiceAccount(defaultUserInfo, googleProject).futureValue
+//  }
+//
+//  def notebookServiceAccount(googleProject: GoogleProject): Option[WorkbenchEmail] = {
+//    serviceAccountProvider.getNotebookServiceAccount(defaultUserInfo, googleProject).futureValue
+//  }
```
Comment (kyuksel) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/TestLeoRoutes.scala]: Remove the commented out lines above

**Review by kyuksel — COMMENTED** [review]
Looks good 👍 Minor comments inline. I'd give @rtitle a chance to look, too in case he can give more meaningful feedback.

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala [hunk 3]
```diff
@@ -166,10 +163,10 @@ class ProxyRoutesSpec extends FlatSpec with BeforeAndAfterAll with BeforeAndAfte
     //  - `upgradeResponse` is a Future[WebSocketUpgradeResponse] that completes or fails when the connection succeeds or fails.
     //  - `result` is a Future[String] with the stream completion from the incoming sink.
     val (upgradeResponse, result) =
-      outgoing
-        .viaMat(webSocketFlow)(Keep.right)
-        .toMat(incoming)(Keep.both)
-        .run()
+    outgoing
```
Comment (rtitle) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala]: I think the indentation before was probably correct?
```
val (foo, bar) = 
  baz
```

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/AuthProviderSpec.scala [hunk 4]
```diff
@@ -3,69 +3,35 @@ package org.broadinstitute.dsde.workbench.leonardo.service
...
-import net.ceedubs.ficus.Ficus._
-import org.broadinstitute.dsde.workbench.google.GoogleStorageDAO
-import org.broadinstitute.dsde.workbench.leonardo.auth.sam.MockPetClusterServiceAccountProvider
-import org.broadinstitute.dsde.workbench.leonardo.dao.google.MockGoogleComputeDAO
-import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
-import org.broadinstitute.dsde.workbench.leonardo.util.BucketHelper
-import org.mockito.Mockito
 
-import scala.concurrent.Future
+class AuthProviderSpec extends FreeSpec with ScalatestRouteTest with Matchers with MockitoSugar with TestComponent with ScalaFutures with OptionValues with GcsPathUtils with TestProxy with BeforeAndAfterAll with CommonTestData{
```
Comment (rtitle) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/AuthProviderSpec.scala]: Thank you - it always bothered me that this test didn't extend `CommonTestData`

**Review by rtitle — APPROVED** [review]
Looks good!

I noticed the mlr test failed -- you should probably rebase to get a fix for that in dev.

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala (hunk 3 above)
Comment (vkumra-broad) [reply src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/ProxyRoutesSpec.scala]: IntelliJ Auto Indent 👎

**Pull request merged**